lib/MusicBrainz/Server/WebService/XMLSerializer.pm

```diff
@@ -1322,16 +1336,36 @@ sub _serialize_tag
 
 sub _serialize_genre_list
 {
-    my ($self, $parent_node, $entity, $inc, $stash) = @_;
+    my (
+        $self,
+        $parent_node,
+        $list,
+        $inc,
+        $stash,
+        $toplevel,
+        $genre_counts,
+        $skip_attributes
+    ) = @_;
+
     return if $in_relation_node;
 
-    my $opts = $stash->store($entity);
     my $list_node = $parent_node->addNewChild(undef, 'genre-list');
 
-    foreach my $tag (sort_by { $_->tag->name } @{$opts->{genres}})
+    # No attributes on entities' genre-list, for parity with tag-list
+    unless ($skip_attributes) {
+        set_list_attributes($list_node, $list);
+    }
```

Comment:
If `$toplevel` is set correctly I think you can just use that and remove this flag.